Added StereoVision clibration tools and depth_map_test module

LucasSwitz opened the pull request.

LucasSwitz committed:
Added StereoVision clibration tools and depth_map_test module

ScottSWu commented:
You typically don't want to commit an entire other repository to this one. What is the StereoVision repo used for? There may be a better way to reference it.

LucasSwitz commented:
It basically just wraps OpenCV's calibration functions in scripts that will
capture and analyze our checkerboard photos and output numpy files that
contain things like a transform matrix and focal length. I could just link
to the StereoVision repo in the readme and explain what it was used for.

On Feb 13, 2017 11:26 AM, "Scott Wu" <notifications@github.com> wrote:

> You typically don't want to commit an entire other repository to this one.
> What is the StereoVision repo used for? There may be a better way to
> reference it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cornell-cup/cs-slam-test/pull/1#issuecomment-279442542>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGFU7NXLpb4K7V9wzOTtd9VS55trnFqMks5rcIQqgaJpZM4L-yB6>
> .
>

ScottSWu commented:
Yeah that would be better. It looks like you can even install it through `pip`.

LucasSwitz committed:
Removed StereoVision and updated Readme

I removed the StereoVision Library from this git and instead provided a
link to it in the README.

ScottSWu commented on README.md:
```diff
@@ -0,0 +1,11 @@
+
+###Tools Used
+
+[StereoVision](https://github.com/erget/StereoVision) : OpenCV wrapper for simple calibration of stereo cameras. Contains wrapper modules as well as scripts for quick calibration.
```
Perhaps you want to give instructions / snippets of code on how to calibrate with stereovision, since it's not directly clear how to do so.

ScottSWu commented on stereo/depth_map_test.py:
```diff
@@ -0,0 +1,54 @@
+
+def get_depth(img_1, img_2):
+
+
```
Let's be clean and remove extra whitespace

ScottSWu commented on stereo/depth_map_test.py:
```diff
@@ -0,0 +1,54 @@
+
+        _, img_2 = vc_2.read()
+
+
```
whitespace here too

ScottSWu commented on stereo/depth_map_test.py:
```diff
@@ -0,0 +1,54 @@
+    stereo = cv2.StereoBM_create(numDisparities=16, blockSize=15)
+    disparity = stereo.compute(img_1_grey, img_2_grey)
+
+    return disparity.astype("uint8")
```
We need the `astype("uint8")` so we can display it with `imshow`. However if we're using depth for SLAM, we typically want the full precision computed from stereo matching. My suggestion would be to move the `astype` down into `imshow` so that `get_depth` does not lose any precision.

LucasSwitz committed:
Cleaned Code and added precision to get_depth

- changed get_depth() to return the 16bit float array rather than the
converted uint8 array

- removed unused imports and whitespace

ScottSWu commented:
Looks good! Go ahead and merge.
